PLBenchmarks/ligands.py

```diff
@@ -0,0 +1,320 @@
+                [index[0].tolist(), index[1].tolist()]
+            )
+            for obs in self._observables:
+                if ("measurement", obs) in list(self._data.index):
+                    self._data = self._data.append(
+                        pd.Series(
+                            [0],
+                            index=pd.MultiIndex.from_tuples(
+                                [("measurement", f"e_{obs}")]
+                            ),
+                        )
+                    )
+                    values = self._data[("measurement", f"{obs}")]
+                    # let pint figure out what the unit means
+                    u = utils.unit_registry(values[2])
+                    self._data[("measurement", f"e_{obs}")] = values[1] * u
+                    self._data[("measurement", obs)] = values[0] * u
+
+    def derive_observables(
+        self,
+        derived_type="dg",
+        destination="DerivedMeasurement",
+        out_unit=utils.unit_registry("kcal / mole"),
+    ):
+        """
+        Derive observables from (stored) primary data, which is then stored in the :py:class:`pandas.DataFrame`
+
+        :param derived_type: type of derived observable, can be any of 'dg' 'ki', 'ic50' or 'pic50'
+        :param destination: string with column name for 'pandas.DataFrame' where the derived observable should be stored.
+        :param out_unit: unit of type :py:class:`pint` unit of derived coordinate
+        :return: None
+        """
+        assert (
+            derived_type in self._observables
+        ), "Observable to be derived not known. Should be any of dg, ki, ic50, or pic50"
+        for obs in self._observables:
+            if ("measurement", obs) in list(self._data.index):
+                self._data = self._data.append(
+                    pd.Series(
+                        [
+                            utils.convert_value(
+                                self._data[("measurement", obs)],
+                                obs,
+                                derived_type,
+                                out_unit=out_unit,
+                            ),
+                            utils.convert_error(
+                                self._data[("measurement", f"e_{obs}")],
+                                self._data[("measurement", obs)],
+                                obs,
+                                derived_type,
+                                out_unit=out_unit,
+                            ),
+                        ],
+                        index=pd.MultiIndex.from_tuples(
+                            [
+                                (destination, derived_type),
+                                (destination, f"e_{derived_type}"),
+                            ]
+                        ),
+                    )
+                )
+                break
+        else:
+            print(f"Conversion to observable {derived_type} not possible.")
```

Comment:
It may be useful to raise an exception here, depending on whether or not you want this function to fail on an impossible conversion. Printing alone does not always indicate the error, such as if another function calls this function without this one erroring out.